ext/standard/type.c

```diff
@@ -141,9 +145,36 @@ PHP_FUNCTION(intval)
 
 	if (Z_TYPE_P(num) != IS_STRING || base == 10) {
 		RETVAL_LONG(zval_get_long(num));
-	} else {
-		RETVAL_LONG(ZEND_STRTOL(Z_STRVAL_P(num), NULL, base));
+		return;
+	}
+	
+	strval = Z_STRVAL_P(num);
+	strlen = Z_STRLEN_P(num);
+
+	while (isspace(*strval) && strlen) {
+		strval++;
+		strlen--;
 	}
+
+	/* Length of 3+ covers "0b#" and "-0b" (which results in 0) */
+	if (strlen > 2) {
+		tmpval = strval;
+		if (tmpval[0] == '-' || tmpval[0] == '+') {
+			tmpval++;
+		}
+
+		if (tmpval[0] == '0' && tmpval[1] == 'b') {
```

Comment:
Do we accept `0X` next to `0x` in intval? If so, `0B` should be supported here as well.